setuptools/build_meta.py

```diff
@@ -385,7 +385,8 @@ def _build_with_temp_dir(self, setup_command, result_extension,
 
         # Build in a temporary directory, then copy to the target.
         os.makedirs(result_directory, exist_ok=True)
-        with tempfile.TemporaryDirectory(dir=result_directory) as tmp_dist_dir:
+        temp_opts = {"prefix": ".tmp-", "dir": result_directory}
```

Comment:
The temp directory here is created inside `result_directory` because this is the only place we can be sure the current process has OS-level rights to write to the disk...